### ISSUE TITLE
non `Int32` categorical references

### ISSUE THREAD

**Issue by ExpandingMan** [JuliaData/Feather.jl#80]
This issue is to discuss how we deal with categorical references that are not `Int32`.  This is a violation of the Arrow [standard](https://github.com/apache/arrow/blob/master/format/Layout.md) (this is not the only suspicious thing that the Feather format does).  In Julia it is very easy to support references of any type, but my main concern is about consistency with other arrow implementations.  Initially I thought that it was *only* Feather.jl that produced non `Int32` references, but python feather seems to deal with them just fine, so I'm not sure what the real source of this is.

It does seem that some people have some data sitting around that actually does use non `Int32` references.

My suggested solution for the time being is that we support *reading* non `Int32` references, but for *writing* we only ever use `Int32`.  I'm going to make the appropriate changes to Arrow.jl so that #78 implements this.

**Comment by quinnj** [comment]
I agree w/ that approach; support reading, but only produce valid Arrow.

**Comment by sglyon** [comment]
I also agree! Writing paper arrow, but reading stuff in the wild is a great balance
